torchtext/datasets/text_classification.py

```diff
@@ -27,27 +27,36 @@
 }
 
 
-# TODO: Replicate below
-#  tr '[:upper:]' '[:lower:]' | sed -e 's/^/__label__/g' | \
-#    sed -e "s/'/ ' /g" -e 's/"//g' -e 's/\./ \. /g' -e 's/<br \/>/ /g' \
-#        -e 's/,/ , /g' -e 's/(/ ( /g' -e 's/)/ ) /g' -e 's/\!/ \! /g' \
-#        -e 's/\?/ \? /g' -e 's/\;/ /g' -e 's/\:/ /g' | tr -s " "
-_normalize_pattern_re = re.compile(r'[\W_]+')
+_normalize_pattern_re = [re.compile(r'\''), re.compile(r'\"'),
+                         re.compile(r'\.'), re.compile(r'<br \/>'),
+                         re.compile(r','), re.compile(r'\('),
+                         re.compile(r'\)'), re.compile(r'\!'),
+                         re.compile(r'\?'), re.compile(r'\;'),
+                         re.compile(r'\:'), re.compile(r'\:'),
+                         re.compile(' +')]
+
+replaced_string = [' \'  ', '',
+                   ' . ', ' ',
+                   ' , ', ' ( ',
+                   ' ) ', ' ! ',
+                   ' ? ', ' ',
+                   ' ', ' ',
+                   ' ']
 
 
 def text_normalize(line):
     """
     Basic normalization for a line of text.
     Normalization includes
     - lowercasing
-    - replacing all non-alphanumeric characters with whitespace
+    - complete some basic text normalization for English words.
```

Comment:
I think it's a good idea to explicitly list the normalizations that are applied so that the user knows exactly what's going to happen without having to look at the code.